backend/open_webui/utils/oauth.py

```
@@ -238,206 +240,336 @@ async def login(self, request, provider):
+                
+            return resp.json()
+        except Exception as e:
+            log.error(f"Error getting user info for {provider_name}: {str(e)}")
+            log.exception(e)
+            return None
 
-            # If not found by oauth_sub, try to find by email
-            if not user:
-                user = Users.get_user_by_email(email)
-                log.error(f"Looking for user by email: {email}, found: {user is not None}")
+    async def handle_callback(self, request, provider_name, response=None):
```

Ruff (0.8.2)

Missing return type annotation for public function `handle_callback`

(ANN201)